pkg/buffer/factory.go

```diff
@@ -0,0 +1,72 @@
+package buffer
+
+import (
+	"io"
+	"sync"
+
+	"github.com/pion/transport/packetio"
+)
+
+type Factory struct {
+	sync.RWMutex
+	videoPool   *sync.Pool
+	audioPool   *sync.Pool
+	rtpBuffers  map[uint32]*Buffer
+	rtcpReaders map[uint32]*RTCPReader
+}
+
+func NewBufferFactory() *Factory {
+	return &Factory{
+		videoPool: &sync.Pool{
+			New: func() interface{} {
+				return NewBucket(2*1000*1000, true)
```

Comment:
could you add a comment about the intuition around this constant and the others in the pr?